src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -232,12 +232,29 @@ private void RunChildren()
         private void CreateChildWorkItems()
         {
             _children = new List<WorkItem>();
-
+            
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
                     _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+            Sort();
         }
 
+        /// <summary>
+        /// Sorts tests under this suite.
+        /// </summary>
+        private void Sort()
+        {
+                this._children.Sort(delegate (WorkItem x, WorkItem y)
+                {
+                    var xKey = 0;
+                    var yKey = 0;
+                    if (x.Test.Properties.ContainsKey(PropertyNames.Order)) int.TryParse(x.Test.Properties[PropertyNames.Order][0].ToString(), out xKey);
+                    if (y.Test.Properties.ContainsKey(PropertyNames.Order)) int.TryParse(y.Test.Properties[PropertyNames.Order][0].ToString(), out yKey);
+                    return xKey.CompareTo(yKey) != 0
+                    ? xKey.CompareTo(yKey) : x.Test.Name.CompareTo(y.Test.Name); ;
```

Comment:
Don't sort by test name here. It has nothing to do with work item order. Just run all ordered items first followed by ordered items. You may want to initialize the keys to MaxValue for that purpose.